src/lib/sap_ha/configuration/cluster.rb

```diff
@@ -278,6 +278,16 @@ def other_nodes_ext
         end.compact
       end
 
+      def get_primary_on_primary
+        SapHA::System::Network.ip_addresses.each do |my_ip|
+	  @nodes.map do |_, node|
```

Comment:
This `map` call is useless, the result is not (or should not as it seems) used as the result of this function. You can use `each` here.

Reply:
Thanks